### PR TITLE
Python wal acceptor tests

### PR DESCRIPTION
**Pull request by arssher**
Includes fixtures for wal acceptors and associated setup.
    
    Nothing really new here, but surprisingly this caught some issues in
    walproposer.

### CODE REVIEW

**Comment by funbringer** [comment]
So @arssher could you describe your experience with mypy's type system in a few words? Any downsides? I'd certainly take them into account.

**Comment by arssher** [comment]
Well, it was easy to put out types following your examples (the only question was whether to put the type in quotes, but SO as always ready to help). However, I haven't set up language server for Python yet and to me this codebase is not so large/complex so that types could provide tremendous help...

**Comment by kelvich** [comment]
LGTM, let's merge this

File: test_runner/batch_others/test_wal_acceptor.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        pg.safe_psql("CREATE TABLE t(key int primary key, value text)")
+
+    # Populate data
+    for pg in pgs:
+        pg.safe_psql("INSERT INTO t SELECT generate_series(1,100000), 'payload'")
+
+    # Check data
+    for pg in pgs:
+        res = pg.safe_psql("SELECT sum(key) FROM t")
+        assert res[0] == (5000050000, )
```
Comment (funbringer) [inline test_runner/batch_others/test_wal_acceptor.py:56]: Could we merge those 3 loops into one, or is there a specific reason (perhaps, some ordering/interleaving concerns) to keep them separate? I'd like to see a comment in the latter case.

File: test_runner/batch_others/test_wal_acceptor.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    for i in range(n_inserts):
+        cur.execute("INSERT INTO t values (%s, 'payload');", (i + 1, ))
+
+        if random.random() <= fault_probability:
+            if failed_node is None:
+                failed_node = wa_factory.instances[random.randrange(0, n_acceptors)]
+                failed_node.stop()
+            else:
+                failed_node.start()
+                failed_node = None
```
Comment (funbringer) [inline test_runner/batch_others/test_wal_acceptor.py:90]: As has been discussed in private: overall, I'd like to see more comments outlining test design decisions and goals.

Comment (arssher) [reply test_runner/batch_others/test_wal_acceptor.py:92]: added the comment

File: test_runner/batch_others/test_wal_acceptor.py (hunk 1 above)
Comment (arssher) [reply test_runner/batch_others/test_wal_acceptor.py:58]: Added the comment.

File: test_runner/batch_others/test_wal_acceptor.py (hunk 2 above)
Comment (arssher) [reply test_runner/batch_others/test_wal_acceptor.py:92]: Tried to do it a bit better.

**Pull request merged**